backend/layers/processing/upload_failures/app.py

```diff
@@ -161,9 +161,11 @@ def cleanup_artifacts(dataset_id: str, error_step_name: str) -> None:
     with logger.LogSuppressed(Exception, message="Failed to clean up artifacts."):
         object_key = os.path.join(os.environ.get("REMOTE_DEV_PREFIX", ""), dataset_id).strip("/")
         if not error_step_name or error_step_name == "download-validate":
-            delete_many_from_s3(os.environ["ARTIFACT_BUCKET"], object_key)
+            delete_many_from_s3(os.environ["ARTIFACT_BUCKET"], object_key + "/")
+            delete_many_from_s3(os.environ["DATASET_BUCKET"], object_key + ".")
+
         if not error_step_name or error_step_name == "cxg":
             cellxgene_bucket = os.getenv(
                 "CELLXGENE_BUCKET", default=f"hosted-cellxgene-{os.environ['DEPLOYMENT_STAGE']}"
             )
-            delete_many_from_s3(cellxgene_bucket, object_key)
+            delete_many_from_s3(cellxgene_bucket, object_key + ".cxg/")
```

Comment:
@Bento007  Shouldn't we also be deleting from the artifact and dataset buckets when the error step is "cxg"?

If the processing pipeline errors, we should delete everything. So, since download-validate is first, we only have to worry about the buckets that get files uploaded to them during that step. But if cxg fails, we should delete from buckets that get files during download-validate OR cxg.

Reply:
That makes sense.